tools/clang/tools/dxcvalidator/dxcvalidator.cpp

```diff
@@ -47,12 +48,50 @@ struct DiagRestore {
   ~DiagRestore() { Ctx.setDiagnosticHandler(OrigHandler, OrigDiagContext); }
 };
 
+static void HashAndUpdate(DxilContainerHeader *Container) {
+  // Compute hash and update stored hash.
+  // Hash the container from this offset to the end.
+  static const uint32_t DXBCHashStartOffset =
+      offsetof(struct DxilContainerHeader, Version);
+  const unsigned char *DataToHash =
+      (const unsigned char *)Container + DXBCHashStartOffset;
+  unsigned AmountToHash = Container->ContainerSizeInBytes - DXBCHashStartOffset;
+  ComputeHashRetail(DataToHash, AmountToHash, Container->Hash.Digest);
+}
+
+static void HashAndUpdateOrCopy(uint32_t Flags, IDxcBlob *Shader,
+                                IDxcBlob **Hashed) {
+  if (Flags & DxcValidatorFlags_InPlaceEdit) {
```

Comment:
If we were to add a flag to skip hashing and container modification called `DxcValidatorFlags_SkipHash`, we could implement it like so:

```suggestion
  if (Flags & DxcValidatorFlags_SkipHash) {
    *Hashed = Shader;
    Shader->AddRef();
  } else if (Flags & DxcValidatorFlags_InPlaceEdit) {
```